### ISSUE TITLE
Windows command line: sclang only finds the server if started from sclang-folder

### ISSUE THREAD

**Issue by bagong** [supercollider/supercollider#2719]
As on other OSses sclang.exe and scide.exe reside in the same application folder on Windows. On Windows though, if sclang is not started from the sclang folder (working directory), sclang will not find the server. So if you are in a setting where this can start sclang:

./build/install/SuperCollider/sclang.exe

sclang started this way cannot boot the server (error: command scsynth.exe not known). A situation in which you often meet this problem is after a fresh build, if you run sc in an IDE and the working directory is not set to the sclang-directory. And of course it makes command-line use of sclang inflexible. In normal sc-ide use the problem is masked because the OS sets the working directory to the sclang directory.

This has most likely to do with the way resourceDir is created in SC_DirUtils on Windows.

**Comment by bagong** [comment]
It is not the ResourceDirectory in SC_DirUtils, which is actually implemented the right, flexible way. Also Platform.resourceDir returns even paths with space correctly. It is the Server.program-command in Platform/windows/WindowsPlatform.sc. I've fixed this, but as the fix also includes dealing with spaces I will try to catch other cases of badly handled spaces in the path before I post a PR. Hopefully tonight. Hopefully similar, all that is needed is a .quote in the right place.

**Comment by brianlheim** [comment]
Excellent, thank you for tracking that down!

**Comment by erikgolts** [comment]
I am still seeing this behavior as of 3.10.0.  Are there any known workarounds?

**Comment by bagong** [comment]
Yes, quite simple actually. The term used by sclang to boot the server is stored in `SCClassLibrary\Platform\Windows\WindowsPlatform.sc`. It's the assignment to `Server.program`. For me, replacing `"scsynth.exe"`  by `(Platform.resourceDir +/+ "scsynth.exe").quote` does the trick. The `quote` is used to wrap the term in quotation marks, which prevents that a likely space in the path, e.g. in "Program Files", becomes a problem. Sorry for the late reply, I don't follow regularly these days...

**Comment by erikgolts** [comment]
@bagong thanks for that and no worries on the late reply. I gave it a try and it allowed me to at least be able to run `postln` successfully in a .scd file.  But when I try to `s.boot` in the same .scd file I get:

'C:\SuperCollider\scsynth.exe" -u 57110 -a 1024 -i 2 -o 2 -b 1026 -H "MME' is not recognized as an internal or external command,
operable program or batch file.

It seems like it's trying to run Device options list as a command, as MME is the first device it finds. 

As a test, I removed the `().quote` and just put `Platform.resourceDir +/+ "scsynth.exe";` in, and then it worked fine, booting a server and happily generating a sine tone.  So for reference, in `SCClassLibrary\Platform\Windows\WindowsPlatform.sc` I changed `Server.program` to equal `Platform.resourceDir +/+ "scsynth.exe";`

**Comment by bagong** [comment]
@erikgolts I think the main info was that you could add a command in that Platform file under Server.program. I am glad you found something for your case. Fortunately you don't have a space in your path, I guess it would be a problem, if you had. But I am not entirely sure, some things have changed in the latest SC-versions, it might be different now. There is another possibility, btw, that might also be worth some experimenting: in the OS-part of the command you could use "start", and for the quotation SC's .format() might be useful. `start` has Windows system flags (e.g. process-priority, in theory there is even "realtime") and provides a slot at the end for arguments. I currently use the lines below and haven't run into problems with added/adding parameters. Would be nice if somebody invested some time to find the best command line that covers all cases well. The current state is pretty ugly. I think it's a remnant of hasty Windows-development.

````
var srvProg = Platform.resourceDir.quote;
Server.program = "start /B /HIGH /D % scsynth.exe".format(srvProg);
`````

**Comment by madskjeldgaard** [comment]
> 
> 
> Yes, quite simple actually. The term used by sclang to boot the server is stored in `SCClassLibrary\Platform\Windows\WindowsPlatform.sc`. It's the assignment to `Server.program`. For me, replacing `"scsynth.exe"` by `(Platform.resourceDir +/+ "scsynth.exe").quote` does the trick. The `quote` is used to wrap the term in quotation marks, which prevents that a likely space in the path, e.g. in "Program Files", becomes a problem. Sorry for the late reply, I don't follow regularly these days...
> 
> Btw, I don't know why yet, but this seems to work in 3.10 only, not in 3.9...

This solution helped me get scsynth working on the Ubuntu Linux subsystem WSL in Windows 10. Not sure I understand why but it seems to work.

**Comment by hosuaa** [comment]
still an issue 3/8/2022 on windows.

**Comment by meiry** [comment]
@dyfer 
i can confirm , when it will be released ?

**Comment by dyfer** [comment]
Thanks @meiry for confirming.
This will be likely released in SC 3.14. There's no timeline at this point, sorry.
